dll/win32/syssetup/wizard.c

```diff
@@ -355,11 +355,170 @@ AckPageDlgProc(HWND hwndDlg,
                     PropSheet_SetWizButtons(GetParent(hwndDlg), PSWIZB_BACK | PSWIZB_NEXT);
                     if (pSetupData->UnattendSetup)
                     {
+                        SetWindowLongPtr(hwndDlg, DWLP_MSGRESULT, IDD_PLATFORM);
+                        return TRUE;
+                    }
+                    break;
+
+                case PSN_WIZBACK:
+                    pSetupData->UnattendSetup = FALSE;
+                    break;
+
+                default:
+                    break;
+            }
+        }
+        break;
+
+        default:
+            break;
+    }
+
+    return FALSE;
+}
+
+static BOOL
+WritePlatformOption(PLATFORM_OPTION nOption)
+{
+    return TRUE;
+    //static const WCHAR s_szPlatformOptions[] = L"System\\CurrentControlSet\\Control\\ProductOptions";
+    //HKEY hKey;
+    //LONG error;
+    //LPCWSTR pData;
+    //DWORD cbData;
+    //
+    //error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, s_szPlatformOptions, 0, KEY_WRITE, &hKey);
+    //if (error)
+    //    return FALSE;
+    //
+    //switch (nOption)
+    //{
+    //    case PLATFORM_OPTION_2003_SERVER:
+    //        /* write ProductSuite */
+    //        pData = L"Enterprise\0Terminal Server\0";
+    //        cbData = (lstrlenA("Enterprise") + lstrlenA("Terminal Server") + 3) * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductSuite", 0, REG_MULTI_SZ, (BYTE *)pData, cbData);
+    //        if (error)
+    //            break;
+    //
+    //        /* write ProductType */
+    //        pData = L"ServerNT";
+    //        cbData = (lstrlenW(pData) + 1) * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductType", 0, REG_SZ, (BYTE *)pData, cbData);
+    //        break;
+    //
+    //    case PLATFORM_OPTION_XP:
+    //        /* write ProductSuite */
+    //        pData = L"\0";
+    //        cbData = 2 * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductSuite", 0, REG_MULTI_SZ, (BYTE *)pData, cbData);
+    //        if (error)
+    //            break;
+    //
+    //        /* write ProductType */
+    //        pData = L"WinNT";
+    //        cbData = (lstrlenW(pData) + 1) * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductType", 0, REG_SZ, (BYTE *)pData, cbData);
+    //        break;
+    //}
+    //
+    //RegCloseKey(hKey);
+    //return error == ERROR_SUCCESS;
```

Comment:
Nothing wrong here but next time use `#if 0 {...}` to disable a certain chunk of code to be compiled -- FIY.

Reply:
Uncomment and enabled.